app/models/facility.rb

```diff
@@ -41,8 +40,12 @@ class Facility < ApplicationRecord
 
   friendly_id :name, use: :slugged
 
-  def report_on_patients
-    registered_patients
+  def dashboard_analytics
+    query = FacilityAnalyticsQuery.new(self)
+
+    [query.follow_up_patients_by_month,
+     query.registered_patients_by_month,
+     query.total_registered_patients].compact.inject(&:deep_merge)
```

Comment:
Do we need compact here?
Would any of the three elements be nil?

Reply:
Yeah, the elements can be nil when there are no `follow_up_patients` for example. This is happening for a few districts in SBX data.